src/main/java/teammates/storage/sqlentity/Student.java

```diff
@@ -0,0 +1,200 @@
+package teammates.storage.sqlentity;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Objects;
+
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+
+import com.googlecode.objectify.annotation.Entity;
+import com.googlecode.objectify.annotation.Id;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.ForeignKey;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.util.JsonUtils;
+
+/**
+ * An association class that represents the association Account -->
+ * [enrolled in] --> Course.
+ */
+@Entity
+@Table(name = "Student")
+public class Student extends BaseEntity {
+    // TODO
+    // [X] Map FKs (Done for Student)
+    // [ ] Ancestor User
+    // [ ] Fill other CourseStudent related methods
+    // [ ] Fill sanitizeForSaving and getInvalidityInfo if necessary
+
+    // should this be auto incremented or generateId from CourseStudent
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private int id;
+
+    // note: set @OneToMany(mappedBy = "student") in Team
+    @ManyToOne
+    @JoinColumn(name = "id", foreignKey = @ForeignKey(name = "TEAM_ID_FK"), referencedColumnName = "id", nullable = false)
+    private Team team;
+
+    // to cascade?
+    // note: to place @OneToOne annotation in User class for bidirectional r/s
+    @OneToOne
+    @JoinColumn(name = "id", foreignKey = @ForeignKey(name = "USER_ID_FK"), referencedColumnName = "id", nullable = false)
+    private User user;
```

Comment:
Relationship between user and student is a "is-a" relationship (inheritance). Need to look at inheritance mapping in hibernate:
https://www.baeldung.com/hibernate-inheritance.